spring-integration-http/src/main/java/org/springframework/integration/http/inbound/HttpRequestHandlingEndpointSupport.java

```diff
@@ -505,7 +503,23 @@ protected void onInit() throws Exception {
 
 			Message<?> reply = null;
 			if (this.expectReply) {
-				reply = this.sendAndReceiveMessage(message);
+				try {
+					reply = this.sendAndReceiveMessage(message);
+				}
+				catch (MessageTimeoutException e) {
+					if (this.statusCodeExpression != null) {
+						reply = getMessageBuilderFactory().withPayload(e.getMessage())
+									.setHeader(org.springframework.integration.http.HttpHeaders.STATUS_CODE,
+											evaluateHttpStatus())
```

Comment:
> But, yes, we should include the failedMessage.

Having such a specific case with the `MessageTimeoutException` when we know the `failedMessage` we could use it as `rootObject` to evaluate that `statusCodeExpression`.
Otherwise it is with the pure context of use...